### DISCUSSION TITLE
Adding `nuxt-highcharts` to Nuxt.config modules produces errors with other plugins

### DISCUSSION THREAD

**Discussion by patroza** [richardeschloss/nuxt-highcharts#72] [Q&A] [answered]
Adding `nuxt-highcharts` to Nuxt.config modules produces errors with other plugins

**Comment by patroza** [comment, marked as answer]
In .nuxt/types, Nuxt generates: 
`  InjectionType<typeof import("../../../../node_modules/.pnpm/nuxt-highcharts@3.1.1/node_modules/nuxt-highcharts/lib/plugin").default> &` but this file doesn't exist

To workaround the issue, a patch for the npm package:
```patch
diff --git a/lib/plugin.d.ts b/lib/plugin.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..5298df7f0f673c97fe02c77b016ac9998ade2291
--- /dev/null
+++ b/lib/plugin.d.ts
@@ -0,0 +1,2 @@
+declare const _default: any;
+export default _default;
```

**Reply by richardeschloss** [reply]
Hmm, can this solution be moved into ./lib/types.d.ts somehow?

**Reply by cloudfinch-shivansh** [reply]
I created a script file in scripts folder

`import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

// Handle the __dirname issue in ES modules
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Define the path to the nuxt-highcharts folder inside node_modules
const targetPath = path.resolve(__dirname, '../node_modules/nuxt-highcharts/lib/plugin.d.ts');

// Ensure the directory exists
fs.mkdirSync(path.dirname(targetPath), { recursive: true });

// Define the content of the plugin.d.ts file
const content = `
declare const _default: any;
export default _default;
`;

// Write the file to the target path
fs.writeFileSync(targetPath, content, 'utf8');

console.log('plugin.d.ts file has been created inside nuxt-highcharts/lib in node_modules.');
`

and then add execution of this file in postinstall

"postinstall": "nuxt prepare && node scripts/setup-plugin.js"
